src/main/java/io/strimzi/CertificateWatch.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi;
+
+import io.quarkus.runtime.Quarkus;
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Paths;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+
+/**
+ * Quarkus loads the TLS certificate at startup but does not reload it when it changes. So if the Drain Cleaner runs for
+ * too long, it can happen that the certificate expires and Kubernetes cannot call the webhook. To prevent this, the
+ * CertificateWatch is watching for changes to the TLS certificate files. If any changes are detected, it exits quarkus
+ * to restart the container. This is by default enabled for production with the default path {@code /etc/webhook-certificates/}
+ * (which is also the default path in the deployment). It is by default disabled in development. The configuration
+ * options {@code certificate.watch.enabled} and {@code certificate.watch.path} can be used to customize the default
+ * settings.
+ */
+@ApplicationScoped
+public class CertificateWatch {
+    private static final Logger LOG = LoggerFactory.getLogger(CertificateWatch.class);
+
+    private final boolean enabled;
+
+    private ChangeWatcher watcher;
+
+    /**
+     * Constructs the certificate watch. The Watcher class is created only if the certificate watch is enabled.
+     */
+    public CertificateWatch() {
+        this.enabled = ConfigProvider.getConfig().getOptionalValue("certificate.watch.enabled", Boolean.class).orElse(false);
+
+        if (enabled) {
+            try {
+                String path = ConfigProvider.getConfig().getOptionalValue("certificate.watch.path", String.class).orElse("/etc/webhook-certificates/");
+
+                LOG.info("Creating certificate watch for path {}", path);
+
+                this.watcher = new ChangeWatcher(path, () -> {
+                    LOG.info("Certificate change detected => Drain Cleaner will be restarted");
```

Comment:
Pedantry: Can we guarantee at this point that the drain cleaner will be restarted? Maybe it's more honest to just state what we're about to do, which is exit the process.